### PR TITLE
Retain original sandbox errors (from different JavaScript realms) without coercion

### PR DESCRIPTION
**Pull request by davidjb**
This PR retains original errors being returned from different realms. Currently, this module [wraps any such errors](https://github.com/guigrpa/docx-templates/blob/master/src/jsSandbox.ts#L54) as ```new Error(`${err}`)```, coercing their name/message into a string, losing these as separate fields, and losing the stack trace entirely (it gets effectively replaced when the error is re-thrown). The reason it does this is because `instanceof Error` returns `false` for sandboxed errors when the prototype chain doesn't match; these errors are still Error objects, it's just that they can't be detected with `instanceof`.  

For context, `instanceof` will return false if an object is from a different realm (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_realms), which affects use of sandboxing of JavaScript. When sandboxing is enabled (or `runJs` calls out to a different environment - where I found this issue), `err instanceof Error` returns false, due to a different prototype chain. Likewise, when sandboxing is disabled `err instanceof Error` will always return true since it's the same execution environment.

To fix this issue, this PR retains the original error without wrapping. To do so, it adds an `isError` type guard which checks both `instanceof Error` and the mandatory fields of the Error type, and treats matching errors as `Error` thereafter. (The [suggested fix](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_realms) for the same issue with `instanceof Array` is `Array.isArray` but this doesn't exist for Errors, hence duck-typing).  Relevant tests have been added and any other non-errors thrown (e.g. strings) are still treated in the same way. 

This change affects the formatting of `CommandExeuctionError`, since the string coercion no longer happens automatically, so `${err.name}` has been added back in to `CommandExeuctionError`'s message. Most tests continue to work unchanged, but several have required updating. The benefit is that this messaging is now consistent between sandboxing and unsandboxed execution and double-wrapping is avoided.

### CODE REVIEW

File: src/errors.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import { LoopStatus } from './types';
 
+export function isError(err: unknown): err is Error {
+  return (
+    err instanceof Error ||
+    (typeof err === 'object' && !!err && 'name' in err && 'message' in err)
```
Comment (jjhbw) [inline src/errors.ts:6]: Out of interest, which cases did `err instanceof Error` miss that made you add the second clause (`typeof err === 'object'.. etc`)?

**Comment by jjhbw** [comment]
Awesome work and good find. PR looks great. I have one question but otherwise looks good to merge.

File: src/errors.ts (hunk 1 above)
Comment (davidjb) [reply src/errors.ts:6]: So this is what I was talking about in my second paragraph in https://github.com/guigrpa/docx-templates/pull/355#issue-2185180026 - if the error comes from a different realm, like when sandboxing is enabled, the `instanceof` check actually always returns false. If you pop a breakpoint or `console.log` in at https://github.com/guigrpa/docx-templates/blob/v4.11.4/src/jsSandbox.ts#L54,(and ensure that `noSandbox` is false) you’ll see what I mean. 

In my situation specifically, my execution environment is isolated-vm, but it affects any different sandboxing environment.

Comment (jjhbw) [reply src/errors.ts:6]: Ah, duh! I missed that, sorry. Your explanation is very clear. Merging...

Comment (davidjb) [reply src/errors.ts:6]: Cheers @jjhbw - really appreciate it and all your hard work. Would a new release be possible at some point? Thanks again 🙏

Comment (jjhbw) [reply src/errors.ts:6]: Just pushed a new release. Thanks again!

**Pull request merged**